apps/api-extractor/src/generators/DtsRollupGenerator.ts

```diff
@@ -265,8 +276,17 @@ export class DtsRollupGenerator {
         let trimmed: boolean = false;
         if (AstDeclaration.isSupportedSyntaxKind(child.kind)) {
           childAstDeclaration = collector.astSymbolTable.getChildAstDeclarationByNode(child.node, astDeclaration);
+          const childDeclarationMetadata: DeclarationMetadata = collector.fetchMetadata(childAstDeclaration);
+          const symbolMetadata: SymbolMetadata = collector.fetchMetadata(childAstDeclaration.astSymbol);
+          // If we have lower-release method overloads, include that information
+          const useDeclarationReleaseTag: boolean = (
+            childAstDeclaration.declaration.kind === ts.SyntaxKind.MethodDeclaration &&
+            !!childDeclarationMetadata
```

Comment:
@D4N14L Until now, API Extractor has assumed that the effective release tag must be the same for all `AstDeclaration`'s that share a common `AstSymbol`.  This PR is changing that assumption.

By **"effective release tag"** I mean the calculated result that determines the .d.ts rollup and website docs ([SymbolMetadata.releaseTag](https://github.com/microsoft/web-build-tools/blob/bdc77711a5546f7f4681f72a82d4a5769719e4e0/apps/api-extractor/src/collector/SymbolMetadata.ts#L7)).  Whereas the **"declared release tag"** is simply the TSDoc tag found in the comment ([DeclarationMetadata.declaredReleaseTag](https://github.com/microsoft/web-build-tools/blob/bdc77711a5546f7f4681f72a82d4a5769719e4e0/apps/api-extractor/src/collector/DeclarationMetadata.ts#L26)).  When the "declared release tag" is absent, the effective value is inherited from the parent.

Anyway, your PR breaks this assumption for function/member overloads, and #972 suggests other ways it may be further broken in the future.  The .api.json file already anticipates this possibility by storing separate `releaseTag` values for each declaration.

I think we need to adjust the metadata representation a bit to make this more clear.  This should also eliminate the awkward checks for `ts.SyntaxKind.FunctionDeclaration` and `ts.SyntaxKind.MethodDeclaration` in each of the generators.

Maybe it should be like this:
```ts
export class DeclarationMetadata {
  /**
   * This is the release tag that was explicitly specified in the original doc comment, if any.
   */
  public declaredReleaseTag: ReleaseTag = ReleaseTag.None;

  /**
   * The "effective" release tag is a normalized value that is based on `declaredReleaseTag`,
   * but may be inherited from a parent, or corrected if the declared value was somehow invalid.
   * When actually trimming .d.ts files or generating docs, API Extractor uses the "effective" value
   * instead of the "declared" value.
   */
  public effectiveReleaseTag: ReleaseTag = ReleaseTag.None;  // <=== new
  . . .
```

The generators would then use `DeclarationMetadata.effectiveReleaseTag` instead of `SymbolMetadata.releaseTag`.  And if the release tag is missing, it would be inherited from the parent declaration (i.e. `AstDeclaration.parent` instead of `AstDeclaration.astSymbol.parentAstSymbol`).

This approach can also set us up to later handle cases like this:
```ts
/** @public */
enum MergedDeclaration {
    // inherits the "@public" release tag
    One
}

/** @beta */
namespace MergedDeclaration {
    // inherits the "@beta" release tag
    export function parseEnum(value: string): MergedDeclaration {
        return MergedDeclaration.One;
    }
}
```

Let me know if you have any questions about this change.  I'm also reachable on [Gitter](https://gitter.im/octogonz).